drivers/include/sps30.h

```diff
@@ -0,0 +1,254 @@
+/*
+ * Copyright (C) 2020 HAW Hamburg
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser General
+ * Public License v2.1. See the file LICENSE in the top level directory for more
+ * details.
+ */
+
+/**
+ * @defgroup    drivers_sps30 Sensirion SPS30 Particulate Matter Sensor
+ * @ingroup     drivers_sensors
+ * @ingroup     drivers_saul
+ * @brief       Device driver for the Sensirion SPS30 Particulate Matter Sensor
+ *
+ * About
+ * =====
+ *
+ * This driver provides an interface for the Sensirion SPS30 Sensor.
+ * For now only I2C mode is supported.
+ * I2C speed must be set to standard mode (100 kbit/s)
+ *
+ * Wiring
+ * ======
+ *
+ * In ASCII-land the connector side of the sensor would look like this:
+ *
+ *  +------------------------------------------------------------------------
+ *  |                                             ____________________       |
+ *  |                                          __|                    |__    |
+ *  |                                         |__  (1) (2) (3) (4) (5) __|   |
+ *  |                                            |                    |      |
+ *  |                                            |____________________|      |
+ *  |                                                                        |
+ *  +------------[#]------------[#]------------[#]------------[#]------------+
+ *
+ * The numbers refer to following pin names:
+ *
+ * | Pin Nr. | SPS30 Signal Name | Connect to | Notes                         |
+ * | Pin 1   | VDD               | 5 V        | should be within +-10 %       |
+ * | Pin 2   | I2C_SDA / UART_RX | SDA*       | configured by PARAM_SPS30_I2C |
+ * | Pin 3   | I2C_SCL / UART_TR | SCL*       | configured by PARAM_SPS30_I2C |
+ * | Pin 4   | SEL               | GND        |                               |
+ * | Pin 5   | GND               | GND        |                               |
+ *
+ *   *The SCL and SDA pins of the SPS30 sensor are open drain so they must be
+ *    pulled high. Consider that internal pull resistors might be too weak.
+ *    So using external 10 kOhm resistors is recommended for that.
+ *
+ * @{
+ *
+ * @file
+ * @brief       Driver for the Sensirion SPS30 Particulate Matter Sensor
+ *
+ * @author      Michel Rottleuthner <michel.rottleuthner@haw-hamburg.de>
+ */
+
+#ifndef SPS30_H
+#define SPS30_H
+
+#include "periph/gpio.h"
+#include "periph/i2c.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief   SPS30 device instance
+ */
+typedef struct {
+    i2c_t i2c_dev;                 /**< I2C dev the sensor is connected to */
+} sps30_dev_t;
+
+/**
+ * @brief   SPS30 device parameters
+ */
+typedef struct {
+    i2c_t i2c_dev;                /**< I2C dev the sensor is connected to */
```

Comment:
with the above this could later be changed to `union` to allow for I2C and UART support.

Reply:
somehow